### PR TITLE
Add hardware handshake support for serial port based CAN interfaces

### LINKED ISSUE #398: Specifying handshake for serial port based CAN interfaces
So far we can only pass baudrate to the interface constructor like `/dev/ttyUSB0@115200`. Many adapters run at high baudrates so enabling hardware handshake is a must. There are two issues:

1. what handshake options do we need to support: RTS/CTS, Xon/Xoff etc.
2. depending on the answer to the first issue the syntax must be defined. For example: `/dev/ttyUSB0@115200:rtscts` or something like this.

### PR DESCRIPTION
**Pull request by yegorich**
pySerial module provides 'rtscts' option to handle hardware handshake.
This setting will be extracted from __init__'s kwargs.

Example:

```Python
bus = can.interface.Bus(bustype='slcan',
                                      channel='/dev/ttyUSB0@3000000',
                                      bitrate=1000000,
                                      rtscts=True)
```

Closes #398

### CODE REVIEW

File: can/interfaces/serial/serial_can.py [hunk 1]
```diff
@@ -54,8 +54,9 @@ def __init__(self, channel, baudrate=115200, timeout=0.1, *args, **kwargs):
             raise ValueError("Must specify a serial port.")
 
         self.channel_info = "Serial interface: " + channel
+        rtscts = kwargs.get('rtscts', False)
```
[the remark this answers is not in the document]
Comment (yegorich) [reply can/interfaces/serial/serial_can.py]: pylint was complaining about too many call parameters in `__init__`, so I decided to use `kwargs`.

[the remark this answers is not in the document]
Comment (felixdivo) [reply can/interfaces/serial/serial_can.py]: Well, I don't see the problem with more parameters ... makes it way easier to spot the options and defaults.

[the remark this answers is not in the document]
Comment (yegorich) [reply can/interfaces/serial/serial_can.py]: OK. Will do.

**Comment by yegorich** [comment]
v2: move rtscts argument to the parameter list and also document it.

**Comment by yegorich** [comment]
Are you planning to resolve all 8 remaining issues for 2.3 or the could be an earlier release?

**Comment by felixdivo** [comment]
@hardbyte What do you think?

**Pull request merged**